sys/shell/include/shell/shell.h

```diff
@@ -16,40 +16,90 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+
 #ifndef __SHELL_H__
 #define __SHELL_H__
 
-#include <os/os.h>
-
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-struct os_eventq;
+/** @brief Callback called when command is entered.
+ *
+ *  @param argc Number of parameters passed.
+ *  @param argv Array of option strings. First option is always command name.
+ *
+ * @return 0 in case of success or negative value in case of error.
+ */
+typedef int (*shell_cmd_func_t)(int argc, char *argv[]);
+
+struct shell_param {
+    const char *param_name;
+    const char *help;
+};
+
+struct shell_cmd_help {
+    const char *summary;
+    const char *usage;
+    const struct shell_param *params;
+};
 
-typedef int (*shell_cmd_func_t)(int argc, char **argv);
 struct shell_cmd {
-    char *sc_cmd;
+    const char *sc_cmd;
```

Comment:
just name fields name, func, help    those already are part of struct shell_cmd so this is clear